build.gradle.kts

```
@@ -17,15 +17,33 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 import org.jetbrains.kotlin.gradle.dsl.KotlinJvmCompile
+import java.util.*
 
 plugins {
   kotlin("jvm")
   application
 }
 
+val localProperties = Properties().apply {
+  try {
+    load(File(rootDir, "local.properties").inputStream())
+  }
+  catch (t: Throwable) {
+    println("Can't read local.properties: $t, assuming empty")
+  }
+}
+
+val projectorServerSource: String = localProperties["projectorServerSource"].toString().substringBefore('-')
+val projectorServerVersion: String = localProperties["projectorServerSource"].toString().substringAfter('-')
```

Minor suggestion. We can get rid of explicit types here:

```suggestion
val projectorServerSource = localProperties["projectorServerSource"].toString().substringBefore('-')
val projectorServerVersion = localProperties["projectorServerSource"].toString().substringAfter('-')
```

Also, we can make the properties nullable to differentiate the situations when there is no property and when there is a property with "null" value:

```
val projectorServerSource = (localProperties["projectorServerSource"] as String?)?.substringBefore('-')
val projectorServerVersion = (localProperties["projectorServerSource"] as String?)?.substringAfter('-')
```

(this will require to adjust the suggested line 85 from `"null"` to `null`).